src/Avalonia.Controls/DockPanel.cs

```diff
@@ -75,132 +92,186 @@ public bool LastChildFill
             set => SetValue(LastChildFillProperty, value);
         }
 
+        /// <summary>
+        /// Gets or sets the horizontal distance between the child objects.
+        /// </summary>
+        public double HorizontalSpacing
+        {
+            get => GetValue(HorizontalSpacingProperty);
+            set => SetValue(HorizontalSpacingProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the vertical distance between the child objects.
+        /// </summary>
+        public double VerticalSpacing
+        {
+            get => GetValue(VerticalSpacingProperty);
+            set => SetValue(VerticalSpacingProperty, value);
+        }
+
+
         /// <summary>
         /// Updates DesiredSize of the DockPanel.  Called by parent Control.  This is the first pass of layout.
         /// </summary>
         /// <remarks>
         /// Children are measured based on their sizing properties and <see cref="Dock" />.  
-        /// Each child is allowed to consume all of the space on the side on which it is docked; Left/Right docked
+        /// Each child is allowed to consume all the space on the side on which it is docked; Left/Right docked
         /// children are granted all vertical space for their entire width, and Top/Bottom docked children are
         /// granted all horizontal space for their entire height.
         /// </remarks>
-        /// <param name="constraint">Constraint size is an "upper limit" that the return value should not exceed.</param>
+        /// <param name="availableSize">Constraint size is an "upper limit" that the return value should not exceed.</param>
         /// <returns>The Panel's desired size.</returns>
-        protected override Size MeasureOverride(Size constraint)
+        protected override Size MeasureOverride(Size availableSize)
         {
-            var children = Children;
+            var parentWidth = 0.0;
```

Comment:
I'm not sure why all comments previously present in this method were deleted, is that intended?

Reply:
This is because I've rewritten Measure's logic and the previous comment no longer applies